### PR TITLE
Use patterns in ERTP

### PR DESCRIPTION
**Pull request by erights**

### CODE REVIEW

File: packages/ERTP/src/typeGuards.js [hunk 1]
```diff
@@ -1,16 +1,21 @@
-import { isNat } from '@agoric/nat';
-import { passStyleOf } from '@agoric/marshal';
-import { isKey } from '@agoric/store';
+import { M, matches } from '@agoric/store';
+
+const NatValueShape = M.nat();
+const SetValueShape = M.arrayOf(M.key());
+const AmountValueShape = M.or(NatValueShape, SetValueShape);
```
Comment (FUDCo) [inline packages/ERTP/src/typeGuards.js:5]: Defining an amount value as either a nat OR a set of more-or-less-anything seems kinda ad hoc.  Is the rationale captured somewhere?

**Review by FUDCo — APPROVED** [review]
LGTM

File: packages/ERTP/src/typeGuards.js (hunk 1 above)
Comment (erights) [reply packages/ERTP/src/typeGuards.js:5]: Now documented in store/src/types.js. Resolving.

Comment (erights) [reply packages/ERTP/src/typeGuards.js:5]: Damn. What I documented at that location is something you asked for elsewhere: The rationale for using the *Shape suffix. This is now merged before I noticed the mistake, but I'll rectify it in #4215 , to be merged shortly.

I'm also unresolving this conversation until I do.

Comment (erights) [reply packages/ERTP/src/typeGuards.js:5]: Done at https://github.com/Agoric/agoric-sdk/pull/4215/commits/5c481465a501d6fe3863fe2506ac9f20b08ffc90 . Soon to be merged as part of #4215. Resolving.

**Pull request merged**